invenio_rdm_records/requests/access/models.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Database models related to access requests."""
+
+import uuid
+from datetime import date, datetime
+from secrets import token_urlsafe
+
+from flask import current_app
+from invenio_db import db
+from sqlalchemy_utils import UUIDType
+
+from .permissions import AccessRequestTokenNeed
+
+
+class AccessRequestToken(db.Model):
```

Comment:
should be renamed to something like `GuestAccessRequestToken`, in order to minimize confusion with RATs

Reply:
after a brief discussion with @zzacharo on discord, we agreed to keep the name as is